### PR TITLE
[xbmc] Fixes that have come from the migration to vs2015

### PR DESCRIPTION
**Pull request by Paxxi**
Missing include of iterator
Don't wait on addon update job when no repositories are found
whitespace changes
naming collisions

### CODE REVIEW

**Comment by Paxxi** [comment]
jenkins build this please
@MartijnKaijser a bit late but finally :)

File: xbmc/network/httprequesthandler/python/HTTPPythonWsgiInvoker.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ class CHTTPPythonWsgiInvoker : public CHTTPPythonInvoker
 
 private:
   static std::map<std::string, std::string> createCgiEnvironment(const HTTPPythonRequest* httpRequest, ADDON::AddonPtr addon);
-  static void addWsgiEnvironment(HTTPPythonRequest* request, void* environ);
+  static void addWsgiEnvironment(HTTPPythonRequest* request, void* environment);
```
Comment (Montellese) [inline xbmc/network/httprequesthandler/python/HTTPPythonWsgiInvoker.h]: IIRC this was an issue on another platform. I first had it named `environment` as well but then the build failed on some platform but I don't remember which one.

Comment (Paxxi) [inline xbmc/network/httprequesthandler/python/HTTPPythonWsgiInvoker.h]: jenkins built on all platforms so whatever it was it seems to be fixed now

File: xbmc/guilib/GUIWindowManager.cpp [hunk 2]
```diff
@@ -811,6 +810,9 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
 void CGUIWindowManager::CloseDialogs(bool forceClose) const
 {
   CSingleLock lock(g_graphicsContext);
+  if (m_activeDialogs.empty())
```
Comment (Razzeee) [inline xbmc/guilib/GUIWindowManager.cpp]: This if seems obsolete as the for will have the same thing happening.

Comment (Paxxi) [inline xbmc/guilib/GUIWindowManager.cpp]: It's there to protect against the iterator debug assertion,  it doesn't like empty containers

Comment (Razzeee) [inline xbmc/guilib/GUIWindowManager.cpp]: Maybe worth a comment as someone might think it's unneeded?

Comment (Paxxi) [inline xbmc/guilib/GUIWindowManager.cpp]: Agreed, will fix

**Pull request merged**